internal/repos/webhook_build_handler.go

```
@@ -90,29 +89,24 @@ func (w *webhookBuildHandler) handleKindGitHub(ctx context.Context, logger log.L
 		return errcode.MakeNonRetryable(errors.Wrap(err, "handleKindGitHub: secret generation failed"))
 	}
 
-	if err := addSecretToExtSvc(svc, conn, job.Org, secret); err != nil {
-		return errcode.MakeNonRetryable(errors.Wrap(err, "handleKindGitHub: Marshal failed"))
-	}
-
-	id, err = client.CreateSyncWebhook(ctx, job.RepoName, fmt.Sprintf("https://%s", globals.ExternalURL().Host), secret) // TODO: Add to DB
+	id, err := client.CreateSyncWebhook(ctx, job.RepoName, fmt.Sprintf("https://%s", globals.ExternalURL().Host), secret) // TODO: store the webhook
 	if err != nil {
 		return errors.Wrap(err, "handleKindGitHub: CreateSyncWebhook failed")
 	}
 
-	logger.Info(fmt.Sprintf("Created webhook with ID: %d", id))
+	if err := addSecretToExtSvc(svc, conn, job.Org, secret); err != nil {
+		logger.Error("handleKindGitHub: Webhook created but addSecretToExtSvc failed")
```

I guess in this case we need a retry. If we log and not return an error, the job will be considered completed. That's why we should return an error in this case (but not the `non-retryable` one)

---

That makes sense. And I presume the same will be the same for line 79?

---

yep, nice catch!